### PR TITLE
allow users to choose the network type (mainnet, devnet, etc)

### PR DESCRIPTION
**Pull request by magestrio**
## Type

- [X] Feature
- [ ] Bug fix
- [ ] CI/CD
- [ ] Documentation
- [ ] Other

## Description

Brief description of the changes made.

### CODE REVIEW

File: examples/src/features/create-database.ts [hunk 1]
```diff
@@ -31,23 +31,21 @@ async function run() {
...
 
   const databases = await zkdb
     .fromGlobal()
-    .databases(
-      new QueryBuilder<DatabaseSearch>().where('name', 'eq', DB_NAME).build()
-    );
+    .databases();
 
-  assert(databases[0].databaseName === DB_NAME);
+  console.log(databases);
```
Comment (SangTran-127) [inline examples/src/features/create-database.ts:48]: I think we should need to keep the assert

File: packages/proof-service/src/domain/create-proof.ts [hunk 2]
```diff
@@ -33,13 +42,29 @@ export async function createProof(taskId: string) {
...
     if (!merkleHeight || !appPublicKey) {
       throw new Error('Setting is wrong, unable to deconstruct settings');
     }
 
+    const network = await ModelNetwork.getInstance().findOne({
+      id: networkId,
+      active: true,
+    });
+
+    if (network) {
```
Comment (SangTran-127) [inline packages/proof-service/src/domain/create-proof.ts:57]: You mean if (!network)

File: packages/proof-service/src/domain/create-proof.ts [hunk 3]
```diff
@@ -33,13 +42,29 @@ export async function createProof(taskId: string) {
...
+      throw Error('No active network found');
+    }
+
+    Mina.setActiveInstance(
+      Mina.Network({
+        networkId: network!.id as NetworkId,
+        mina: network!.endpoint,
+      })
+    );
```
Comment (SangTran-127) [inline packages/proof-service/src/domain/create-proof.ts:66]: network checked already so don't need to be check

File: packages/proof-service/src/index.ts [hunk 4]
```diff
@@ -2,16 +2,8 @@ import { DatabaseEngine } from '@zkdb/storage';
 import TaskService from './service/task-service.js';
 import logger from './helper/logger.js';
 import { config } from './helper/config.js';
-import { Mina } from 'o1js';
 
 (async () => {
-  const network = Mina.Network({
```
Comment (SangTran-127) [inline packages/proof-service/src/index.ts:8]: Does it work if you remove the setActiveInstance thing

File: packages/serverless/src/apollo/app/database.ts [hunk 5]
```diff
@@ -36,6 +37,7 @@ const DatabaseCreateRequest = Joi.object<TDatabaseCreateRequest>({
   databaseName,
   merkleHeight: Joi.number().integer().positive().min(8).max(128).required(),
   publicKey,
+  networkId: Joi.string().required()
```
Comment (SangTran-127) [inline packages/serverless/src/apollo/app/database.ts:40]: matching 'devnet' and 'mainnet' as well

File: packages/api/src/graphql/database.ts [hunk 6]
```diff
@@ -105,7 +107,12 @@ export const database = <T>(client: TApolloClient<T>) => ({
   >(client, DATABASE_CHANGE_OWNER, (data) => data.dbChangeOwner),
   create: createMutateFunction<
     boolean,
-    { databaseName: string; merkleHeight: number; publicKey: string },
+    {
+      databaseName: string;
+      merkleHeight: number;
+      publicKey: string;
+      networkId: string;
```
Comment (SangTran-127) [inline packages/api/src/graphql/database.ts:114]: networkId: 'mainnet' | 'devnet'

File: packages/serverless/src/apollo/app/database.ts [hunk 7]
```diff
@@ -22,6 +22,7 @@ export type TDatabaseSearchRequest = {
 export type TDatabaseCreateRequest = TDatabaseRequest & {
   merkleHeight: number;
   publicKey: string;
+  networkId: string
```
Comment (SangTran-127) [inline packages/serverless/src/apollo/app/database.ts:25]: networkId: 'mainnet' | 'devnet'

File: packages/serverless/src/apollo/app/database.ts [hunk 8]
```diff
@@ -84,7 +87,7 @@ extend type Query {
 }
 
 extend type Mutation {
-  dbCreate(databaseName: String!, merkleHeight: Int!, publicKey: String!): Boolean
+  dbCreate(databaseName: String!, merkleHeight: Int!, publicKey: String!, networkId: String!): Boolean
```
Comment (SangTran-127) [inline packages/serverless/src/apollo/app/database.ts:90]: using enum

File: packages/serverless/src/domain/use-case/network.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+export async function fillNetworks() {
+  const modelNetwork = ModelNetwork.getInstance();
+
+  const networks = await (await modelNetwork.find()).toArray();
+
+  if (networks.length === 0) {
+    await modelNetwork.insertMany([
+      {
+        id: 'mainnet',
+        endpoint: 'https://api.minascan.io/node/devnet/v1/graphql',
```
Comment (SangTran-127) [inline packages/serverless/src/domain/use-case/network.ts:12]: this endpoint not mainnet

File: packages/serverless/src/domain/use-case/network.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+        active: true,
+      },
+      {
+        id: 'devnet',
```
Comment (SangTran-127) [inline packages/serverless/src/domain/use-case/network.ts:16]: this endpoint not devnet

File: packages/zkdb/src/sdk/impl/database.ts [hunk 11]
```diff
@@ -87,16 +88,6 @@ export class ZKDatabaseImpl implements ZKDatabase {
     return result.unwrap().map((collection) => collection.name);
   }
 
-  async create(merkleHeight: number, publicKey: PublicKey): Promise<boolean> {
```
Comment (SangTran-127) [inline packages/zkdb/src/sdk/impl/database.ts:90]: Can you explain to me why you removed this function?

Comment (magestrio) [reply packages/zkdb/src/sdk/impl/database.ts:90]: We have two exact function, one in globalContext

File: packages/serverless/src/apollo/app/database.ts (hunk 5 above)
Comment (magestrio) [reply packages/serverless/src/apollo/app/database.ts:40]: Moved to createProof method

File: examples/src/features/create-database.ts (hunk 1 above)
Comment (magestrio) [reply examples/src/features/create-database.ts:51]: done

File: packages/proof-service/src/domain/create-proof.ts (hunk 2 above)
Comment (magestrio) [reply packages/proof-service/src/domain/create-proof.ts]: done

File: packages/proof-service/src/domain/create-proof.ts (hunk 3 above)
Comment (magestrio) [reply packages/proof-service/src/domain/create-proof.ts]: done

File: packages/proof-service/src/index.ts (hunk 4 above)
Comment (magestrio) [reply packages/proof-service/src/index.ts:8]: Moved to create-proof method

File: packages/serverless/src/domain/use-case/network.ts (hunk 10 above)
Comment (magestrio) [reply packages/serverless/src/domain/use-case/network.ts]: done

File: packages/serverless/src/apollo/app/database.ts (hunk 8 above)
Comment (magestrio) [reply packages/serverless/src/apollo/app/database.ts]: done

File: packages/serverless/src/apollo/app/database.ts (hunk 7 above)
Comment (magestrio) [reply packages/serverless/src/apollo/app/database.ts]: done

File: packages/api/src/graphql/database.ts (hunk 6 above)
Comment (magestrio) [reply packages/api/src/graphql/database.ts]: done

**Pull request closed without merging**